### PR TITLE
Added tailoring related proficiencies to container recipes

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

SUMMARY: Content "Adds proficiencies to recipes for containers, mainly worn containers"

#### Purpose of change
The basic proficiency framework was completed in #42476 but we still don't have many of these. At some point most complex actions should be linked to a proficiency, and we need to start expanding the proficiency framework to see the gaps remaining.

#### Describe the solution
Dependent on #44201, won't leave draft until that one is merged.

Adds proficiencies to recipes in armor/containers.json

Audits crafting time for work containers. I was quite generous, I think.

Adds a few minor requirements and things as indicated, such as a new `plant_cordage` requirement for withered plants, straw piles, and bark that can be used as cordage and for weaving

Added a lot of reversibility to these recipes, in part because we're going to need ways to get zippers and things and dismantling backpacks is the best way to do it.

#### Describe alternatives you've considered
Continuing to play in a less fun game where you don't have as complex a learning tree

#### Testing
nonproficient character with good tailoring base skill:

easy things easy: (bindle)
image

mostly easy things mostly easy: (pouch)
image

somewhat hard things somewhat hard: (backpack)
image

Hard things hard: (xl backpack)
image

Disassembling will retrieve items, but could use a review of which items come back. Backpacks give buttons, not zippers?
image

#### Additional context
My reversibility changes in particular could use a quick review.

### CODE REVIEW

File: data/json/proficiencies/misc.json [hunk 1]
```diff
@@ -7,6 +7,22 @@
...
+    "name": { "str": "Basketweaving" },
+    "description": "Forming coarse fibers into shapes.",
+    "can_learn": true,
+    "time_to_learn": "6 h"
+  },
+  {
+    "type": "proficiency",
+    "id": "prof_carving",
+    "name": { "str": "Woodcarving" },
+    "description": "Forming coarse fibers into shapes.",
```
Comment (aoae23) [inline data/json/proficiencies/misc.json:22]: Is this supposed to be the same as the Basketweaving description?

Comment (I-am-Erk) [reply data/json/proficiencies/misc.json:22]: Hmm.

Comment (Qrox) [reply data/json/proficiencies/misc.json:22]: It's technically correct.

**Comment by I-am-Erk** [comment]
This is part 4 of an ongoing project that has now received a lot of reviews, and this particular piece doesn't add anything too new or controversial. As a self-declared dictator, I am going to just merge it now. Please direct your concerns to the compain...

GODDAMN IT ZS.

**Comment by ZhilkinSerg** [comment]
I am the senate!

**Comment by I-am-Erk** [comment]
NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

**Pull request merged**